### PR TITLE
Update to `http` v0.2, `http-body` v0.3, `tower-service` v0.3 and…

### PR DESCRIPTION
**Pull request by tesaguri**
This removes `BodyExt::into_buf_stream` since `tokio-buf` crate was removed since `tokio` 0.2, and consequently, `BodyExt` has no methods now. I think there should be an alternative method like `BodyExt::into_stream` or otherwise `BodyExt` should be removed, but that is beyond the scope of this PR.

### CODE REVIEW

File: tower-http-util/Cargo.toml [hunk 1]
```diff
@@ -21,10 +21,8 @@ Utilities for working with HTTP services.
...
-futures = "0.1.25"
-http = "0.1.16"
-http-body = "0.1"
+http = "0.2"
+http-body = "0.3"
 http-connection = "0.1.0"
-tokio-buf = "0.1.0"
-tokio-io = "0.1"
-tower-service = "0.2.0"
+tokio = "0.2"
```
Comment (LucioFranco) [inline tower-http-util/Cargo.toml]:
```suggestion
tokio = { version = "0.2", features = ["io"] }
```

File: tower-request-modifier/tests/request_modification.rs [hunk 2]
```diff
@@ -3,15 +3,17 @@ extern crate tower_request_modifier;
 extern crate tower_service;
 extern crate tower_test;
 
+use futures_test::task::noop_context;
```
Comment (LucioFranco) [inline tower-request-modifier/tests/request_modification.rs]: we should attempt to use `tokio-test` here, should be possible to get a context, you can peek at the other tower tests for that.

File: tower-http-util/Cargo.toml (hunk 1 above)
Comment (tesaguri) [reply tower-http-util/Cargo.toml]: `tokio` v0.2 does not have have a feature named `io`. Also, ` tower-http-util` only uses `AsyncRead` and `AsyncWrite`, which do not depend on any Cargo features.

**Comment by tesaguri** [comment]
Updated to use `tokio-test`.

File: tower-http-util/Cargo.toml (hunk 1 above)
Comment (LucioFranco) [reply tower-http-util/Cargo.toml]: You're right!

**Pull request merged**